src/ProjectTemplates/test/RazorComponentsTemplateTest.cs

```diff
@@ -42,9 +47,12 @@ private void TestApplication(bool publish)
 
         private void TestBasicNavigation()
         {
+            // Give components.server enough time to load so that it can replace
+            // the prerendered content before we start making assertions.
+            Thread.Sleep(5000);
```

Comment:
This seems a bit sketchy. It might be OK temporarily, but maybe we should extend the connected/disconnected APIs that @pranavkm was working on to provide a connection status value so it can be determined directly.

Reply:
Well, ideally what I need is some API to know that components.server.js finished initializing so that I can do ((IJavasScriptExecutor)Browser).Execute("snippet") and determine it's done initializing.

Reply:
That's a crazy amount of work for these tests, and wouldn't work for Blazor in the future. We should let people know when we are done initializing stuff by for example, simply saving the boot promise inside the window[Blazor] object.